### ISSUE TITLE
Chronograf: error when upgrading from v1.10.3 to v1.10.4

### ISSUE THREAD

**Issue by R290** [influxdata/influxdata-docker#749]
I'm using Chronograf with a named volume mounted to `/var/lib/chronograf`. When upgrading from v1.10.3 to v1.10.4 I get the following error:

``` 
time="2024-05-23T19:17:26Z" level=error msg="Unable to create bolt clientUnable to open boltdb; is there a chronograf already running?  open /var/lib/chronograf/chronograf-v1.db: permission denied"
``` 

I traced the issue to the (assumed) permissions on `/var/lib/chronograf/chronograf-v1.db`. In v1.10.3 `chronograf-v1.db` is owned by `root`:
```
root@f4382be4448d:/var/lib/chronograf# ls -al
total 24
drwxr-xr-x 1 chronograf chronograf    44 May 23 19:16 .
drwxr-xr-x 1 root       root          34 Apr 24 07:54 ..
drwx------ 1 root       root          44 May 23 19:16 backup
-rw------- 1 root       root       32768 May 23 19:29 chronograf-v1.db
```

While a fresh start with v1.10.4 shows `chronograf-v1.db` owned by `chronograf`:
``` 
root@f063bcfc3649:/var/lib/chronograf# ls -al
total 24
drwxr-xr-x 1 chronograf chronograf    32 May 23 19:33 .
drwxr-xr-x 1 root       root          34 May 14 02:55 ..
-rw------- 1 chronograf chronograf 32768 May 23 19:33 chronograf-v1.db
```

This should be easily fixable by manually changing permissions I guess, but for a "patch"  release I'd expect a painless upgrade process. Any thoughts on this? The issue occers with both the alpine and non-alpine images.

**Comment by powersj** [comment]
Hey @R290,

> This should be easily fixable by manually changing permissions I guess, but for a "patch" release I'd expect a painless upgrade process.

Your analysis is correct - we did update the default user from `root` to `chronograf` in #731.  Running as root, especially when we did not need to be, is a security issue. While there were no directly reports of any issue, the preventative change was made to reduce the potential.

If you are unable to make the change, then also a check for the `CHRONOGRAF_AS_ROOT` environment variable. Setting that to true will revert to the previous behavior as well.

> for a "patch" release I'd expect a painless upgrade process

I also agree. In this case, as it was a security change, we made the decision to include this as part of the release. If we had waited to the next minor release, unfortunately I think someone could have said a similar statement as well.

Hopefully that helps clear up the intent behind the change.

Thanks!

**Comment by R290** [comment]
I agree with the reasoning and understand the decision made. Since I'm the fist one to create this issue, I guess there are not that many Chronograf users (left)? For future reference, the commands I used to fix this when upgrading from v1.10.3 to v1.10.4 are listed below.

Debian based image:
```
docker exec _chronograf_container_name_ /bin/sh -c "chown -R chronograf:chronograf /var/lib/chronograf"
```

Alpine based image needs some additional steps like creating the group and user & setting permissions on the folder:
```
docker exec _chronograf_container_name_ /bin/sh -c "addgroup -S chronograf && adduser -S chronograf -G chronograf && chmod 0750 /var/lib/chronograf && chown -R chronograf:chronograf /var/lib/chronograf"
```

**Comment by powersj** [comment]
>  Since I'm the fist one to create this issue, I guess there are not that many Chronograf users (left)? 

Right, that or my guess is that many have not updated yet. I kind of expect many users are not actively updating their kapacitor/chronograf unless they need to.

Thanks for the commands, I'm going to go ahead and close this out.
